Cargo.toml

```diff
@@ -7,7 +7,7 @@ members = ["cli", "lib", "lib/gen-protos", "lib/proc-macros", "lib/testutils"]
 [workspace.package]
 version = "0.27.0"
 license = "Apache-2.0"
-rust-version = "1.84" # NOTE: remember to update CI, contributing.md, changelog.md, and install-and-setup.md
+rust-version = "1.84" # NOTE: remember to update rust-toolchain.toml, CI, contributing.md, changelog.md, and install-and-setup.md
```

Comment:
I see rust-toolchain.toml was removed in https://github.com/jj-vcs/jj/pull/5090/files so I'll remove it here as well